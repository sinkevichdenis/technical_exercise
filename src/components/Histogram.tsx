import React from 'react';
import { StyleProp, StyleSheet, View, ViewStyle } from 'react-native';
import { BarChart } from 'react-native-chart-kit';

import { Colors } from 'styles/colors';
import { DEFAULT_BORDER_RADIUS } from 'styles/consts';


type Props = {
  containerStyle?: StyleProp<ViewStyle>,
  verticalLabelRotation?: number,
  width: number,
  height?: number,
  data: any,
}

const chartConfig = {
  backgroundGradientFrom: Colors.LIGHT_BLUE,
  backgroundGradientTo: Colors.LIGHT_BLUE,
  fillShadowGradientFrom: Colors.BLUE,
  fillShadowGradientTo: Colors.BLUE,
  fillShadowGradientFromOpacity: 1,
  fillShadowGradientToOpacity: 1,
  decimalPlaces: 0,
  color: () => Colors.GRAY,
  labelColor: () => Colors.TEXT_BLACK,
  barPercentage: 0.4,
};

export const Histogram: React.FC<Props> = React.memo(({
  containerStyle,
  verticalLabelRotation = 60,
  width,
  height = 350,
  data,
}) => {
  return (
    <View style={ containerStyle }>
      <BarChart
        style={ styles.graph }
        data={ data }
        width={ width }
        height={ height }
        chartConfig={ chartConfig }
        verticalLabelRotation={ verticalLabelRotation }
        fromZero
        showValuesOnTopOfBars
        showBarTops={ false }
      />
    </View>
  );
});

const styles = StyleSheet.create({
  graph: {
    borderRadius: DEFAULT_BORDER_RADIUS,
  },
});
