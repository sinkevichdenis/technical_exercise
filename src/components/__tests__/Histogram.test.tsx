import { render } from '@testing-library/react-native';
import React from 'react';

import { mockHistogramData } from '__mocks__/histogram.mock';

import { Histogram } from '../Histogram';


describe('Histogram component', () => {
  it('should match snapshot', () => {
    const { toJSON } = render(
      <Histogram
        data={ mockHistogramData }
        width={ 500 }
      />);
    expect(toJSON()).toMatchSnapshot();
  });
});
