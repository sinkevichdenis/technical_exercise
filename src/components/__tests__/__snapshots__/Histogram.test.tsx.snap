// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Histogram component should match snapshot 1`] = `
<View
  testID="histogram"
/>
`;
